homeassistant/components/frontier_silicon/config_flow.py

```diff
@@ -101,7 +110,35 @@ async def async_step_user(
             step_id="user", data_schema=data_schema, errors=errors
         )
 
-    async def _async_step_device_config_if_needed(self) -> FlowResult:
+    async def async_step_ssdp(self, discovery_info: ssdp.SsdpServiceInfo) -> FlowResult:
+        """Process entity discovered via SSDP."""
+
+        device_url = discovery_info.ssdp_location
+
+        speaker_name = discovery_info.ssdp_headers.get(SSDP_ATTR_SPEAKER_NAME)
+        self.context["title_placeholders"] = {"name": speaker_name}
+
+        try:
+            self._webfsapi_url = await AFSAPI.get_webfsapi_endpoint(device_url)
+        except FSConnectionError:
+            return self.async_abort(reason="cannot_connect")
+        except Exception as exception:  # pylint: disable=broad-except
+            _LOGGER.exception(exception)
+            return self.async_abort(reason="unknown")
+
+        # For manually added devices the unique_id is the radio_id,
+        # for devices discovered through SSDP it is the UDN
+        unique_id = discovery_info.ssdp_udn
```

Comment:
What does this UDN look like? Are we sure its unique enough to meet the unique id requirements?

https://developers.home-assistant.io/docs/config_entries_config_flow_handler?_highlight=unique&_highlight=id#unique-id-requirements

Reply:
The UDN contains an UUID, for example: `uuid:3dcc7100-f76c-11dd-87af-00226124ca30`

Reply:
Does `uuid:` need to be stripped from the string to match what `get_radio_id()` returns?

Reply:
Those 2 ids have sadly no relation to each other.

Reply:
So to summarize:
- The UDN is a good unique id when discovered through SSDP
- the radio-id is another good unique id, but can only be accessed after authenticating with the device. We can thus not use the radio-id as the sole unique id, because it would not allow us to differentiate two devices where we are not authenticated.
- UDN and radio-id are not equal.

So it's theoretically possible for a user to add the device twice: once via SSDP discovery, once via manual configuration.

AFAIK, it is not possible for a config entry to have multiple unique id's? Otherwise we could also register the radio-id as a secondary unique id for SSDP discovered devices.

Reply:
We aren't allowed to have two different formats for unique ids for the same integration.

Can the UDN be accessed with authentication as well?

Reply:
I triple checked if a new API call had been discovered in the past year, but sadly not. [The API does expose a UUID](https://github.com/flammy/fsapi/blob/master/FSAPI.md#netremotesysinfodmruuid), but that one is different from the UDN returned in the context of the `urn:schemas-frontier-silicon-com:undok:fsapi:1` SSDP.

Wiresharking did show that that UUID is being passed in the context of the DLNA SSDP, but trying to use that will result in a very ugly/brittle solution I'm afraid.

Reply:
Your other option is to call the api to get the radio id once its discovered via SSDP and set it.

If they changed the default PIN and you can't obtain the unique id, and we don't know if we are going to present the user a flow for something that is already setup, the best option is to abort the flow as they can always set it up manually.

To mitigate the cost of probing, we should abort if a config entry already exists for the host we are about to probe.

Reply:
OK, I've removed support for ssdp discovery of devices with nondefault PIN.

The overhead of the ssdp-flow for existing devices is now 1 call to the device discovery URL to retrieve the webfsapi endpoint. This discovery file is small file (<200 bytes):

```
<netRemote>
<friendlyName>Keukenradio</friendlyName>
<version>ir-mmi-FS2026-0400-0512-Stream94i_V2.14.34c.EX86167-V1.05</version>
<webfsapi>http://192.168.1.123:80/fsapi</webfsapi>
</netRemote>
```